src/simulation/LoadGenerator.cpp

```diff
@@ -999,37 +1660,54 @@ LoadGenerator::checkAccountSynced(Application& app, bool isCreate)
 }
 
 void
-LoadGenerator::waitTillComplete(bool isCreate)
+LoadGenerator::waitTillComplete(GeneratedLoadConfig cfg)
 {
     if (!mLoadTimer)
     {
         mLoadTimer = std::make_unique<VirtualTimer>(mApp.getClock());
     }
     vector<TestAccountPtr> inconsistencies;
-    inconsistencies = checkAccountSynced(mApp, isCreate);
+    inconsistencies = checkAccountSynced(mApp, cfg.isCreate());
+    auto sorobanInconsistencies = checkSorobanStateSynced(mApp, cfg);
 
-    if (inconsistencies.empty())
+    // If there are no inconsistencies and we have generated all load, finish
+    if (inconsistencies.empty() && sorobanInconsistencies.empty() &&
+        cfg.isDone())
     {
         CLOG_INFO(LoadGen, "Load generation complete.");
         mLoadgenComplete.Mark();
-        reset();
+        reset(/*resetSoroban=*/false);
         return;
     }
-    else
+    // If we have an inconsistency, reset the timer and wait for another ledger
+    else if (!inconsistencies.empty() || !sorobanInconsistencies.empty())
     {
         if (++mWaitTillCompleteForLedgers >= TIMEOUT_NUM_LEDGERS)
         {
             CLOG_INFO(LoadGen, "Load generation failed.");
             mLoadgenFail.Mark();
-            reset();
+            reset(/*resetSoroban=*/!sorobanInconsistencies.empty());
             return;
         }
 
         mLoadTimer->expires_from_now(
             mApp.getConfig().getExpectedLedgerCloseTime());
-        mLoadTimer->async_wait(
-            [this, isCreate]() { this->waitTillComplete(isCreate); },
-            &VirtualTimer::onFailureNoop);
+        mLoadTimer->async_wait([this, cfg]() { this->waitTillComplete(cfg); },
+                               &VirtualTimer::onFailureNoop);
+    }
+    // If there are no inconsistencies but we aren't done yet, we have more load
+    // to generate, so schedule loadgen
+    else
+    {
+        // If there are no inconsistencies but we aren't done, we must be in a
+        // two phase mode (soroban setup).
+        releaseAssert(cfg.isSorobanSetup());
```

Comment:
This assert is tripping for me when running soroban_upload mode (after running upgrades in the same mission)

Reply:
Looks like I forgot to add SOROBAN_UPLOAD in one of the GeneratedLoadConfig access functions, fixed.